nomad/structs/structs.go

```diff
@@ -3518,6 +3638,18 @@ type Allocation struct {
 	// PreviousAllocation is the allocation that this allocation is replacing
 	PreviousAllocation string
 
+	// DeploymentID identifies an allocation as being created from a
+	// particular deployment
+	DeploymentID string
+
+	// DeploymentHealth marks whether the allocation has been marked healthy or
+	// unhealthy as part of a deployment. It can be unset if it has neither been
+	// marked healthy or unhealthy.
+	DeploymentHealth *bool
```

Comment:
I have a feeling we may want a richer structure here. Can we make this a struct and have a bool field to start? Just incase we want future annotations.